File: 26-06-2025/p2311.kt
    fun longestSubsequence(s: String, k: Int): Int {
        var sum = 0L
        var z = 0
        for (i in s) {
            if(i=='0') {
                z++
            }
        }
        var i = s.length-1
        while (i>=0 && sum<k) {
            if(s[i]=='1') {
                sum+=Math.pow(2.0, (s.length-i-1).toDouble()).toInt()
                z++
            }
            i--
        }
        if(sum>k) {
            z--
        }
        return z
    }

File: 26-06-2025/p2311.js
var longestSubsequence = function (s, k) {
    let count = 0, j = s.length - 1, i = 0, acc = 0;
    while (j >= 0) {
        let positionNumber = Number(s[j]) * Math.pow(2, i);
        j--;
        i++;
        if (acc + positionNumber > k) continue;
        acc += positionNumber;
        count++;
    }
    return count;
};